sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsClientContext.java

```diff
@@ -87,6 +89,11 @@ public void serialize(DiagnosticsClientContext clientContext, JsonGenerator gene
         private RntbdTransportClient.Options options;
         private String rntbdConfigAsString;
         private ConnectionMode connectionMode;
+        private String machineId;
+
+        public void withMachineId(String machineId) {
+            this.machineId = machineId;
```

Comment:
any possibility that the vmId is empty or null for older VMs? should we only allow non-empty vmId to override the machineId?